### ISSUE TITLE
🐛 [firebase_auth][firebase_messaging] Firebase Dart initialisation crashes on iOS if push notifications capabilities are set

### ISSUE THREAD

**Issue by hazzo** [FirebaseExtended/flutterfire#7752]
## Bug report

**Describe the bug**
A clean flutter (2.8.1) project won't initialise run correctly if `firebase_auth` and `firebase_messaging` are installed with iOS push notifications capabilities added. It's kind of annoying as this is the second similar issue regarding Dart initialisation and firebase messaging not working.

### Steps to reproduce

1. `flutter create firebasetest`
2. `flutter pub add firebase_core`
3. `flutter pub add firebase_messaging`
4. `flutter pub add firebase_auth`
5. `flutterfire configure` (configure firebase with cli)
6. Add **push notifications** and **background modes** capabilities 
7. Add following snippet to main file:
```dart
WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );
```

If step **6** is avoided app works correctly but `firebase_messaging` config would be incomplete. Also if step **4** is avoided app works even with capabilities set. Dart initialisation and firebase plugins configuration it's kind of unstable yet.

### Expected behavior

App should run without crashing.

---

## Additional context

Error when running project with `flutter run`

<details><summary>Click to see error</summary>

```bash
Launching lib/main.dart on Guido’s iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: 5B7F2K8QVV
Running pod install...
Running Xcode build...
Xcode build done.                                           12.1s
Installing and launching...
(lldb) 2022-01-01 19:40:24.693007+0100 Runner[24593:5665764] 8.9.0 - [Firebase/Core][I-COR000012] Could not locate configuration file: 'GoogleService-Info.plist'.
[VERBOSE-2:FlutterObservatoryPublisher.mm(101)] Failed to register observatory port with mDNS with error -65555.
[VERBOSE-2:FlutterObservatoryPublisher.mm(103)] On iOS 14+, local network broadcast in apps need to be declared in the app's Info.plist. Debug and profile Flutter apps and modules host VM services on the local network to support debugging features such as hot reload and DevTools. To make your Flutter app or module attachable and debuggable, add a '_dartobservatory._tcp' value to the 'NSBonjourServices' key in your Info.plist for the Debug/Profile configurations. For more information, see https://flutter.dev/docs/development/add-to-app/ios/project-setup#local-network-privacy-permissions
8.9.0 - [Firebase/Core][I-COR000003] The default Firebase app has not yet been configured. Add `FirebaseApp.configure()` to your application initialization. This can be done in in the App Delegate's application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI). Read more: https://goo.gl/ctyzm8.
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The default FirebaseApp instance must be configured before the default Authinstance can be initialized. One way to ensure this is to call `FirebaseApp.configure()` in the App Delegate's `application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI).'
*** First throw call stack:
(0x1828b904c 0x19af2df54 0x182910180 0x1029c58a0 0x1025213e4 0x104ffa9a8 0x185c25248 0x182529914 0x18252b660 0x182539b60 0x182871cd4 0x18282beac 0x18283f3b8 0x19e1cf38c 0x1851df6a8 0x184f5e7f4 0x10251e708 0x102701a24)
libc++abi: terminating with uncaught exception of type NSException
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00000001b9700964 libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`__pthread_kill:
->  0x1b9700964 <+8>:  b.lo   0x1b9700984               ; <+40>
    0x1b9700968 <+12>: pacibsp 
    0x1b970096c <+16>: stp    x29, x30, [sp, #-0x10]!
    0x1b9700970 <+20>: mov    x29, sp
Target 0: (Runner) stopped.
Error connecting to the service protocol: failed to connect to http://127.0.0.1:53373/
```
</details>

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-x64, locale en-ES)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Android Studio (version 3.6)
[✓] IntelliJ IDEA Ultimate Edition (version 2021.2.3)
[✓] Connected device (1 available)
    ! Error: Guido’s iPhone is busy: Preparing the watch for development via Guido’s iPhone. Xcode will continue when Guido’s iPhone is finished. (code -10)
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.15.1
Flutter SDK 2.8.1
firebaseauthmessaging 1.0.0+1

dependencies:
- cupertino_icons 1.0.4
- firebase_auth 3.3.4 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 1.10.6 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_messaging 11.2.4 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]

dev dependencies:
- flutter_lints 1.0.4 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- async 2.8.2 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- collection 1.15.0
- fake_async 1.2.0 [clock collection]
- firebase_auth_platform_interface 6.1.9 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 3.3.5 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.2.3 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.5.3 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_messaging_platform_interface 3.1.4 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 2.2.5 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- intl 0.17.0 [clock path]
- js 0.6.3
- lints 1.0.1
- matcher 0.12.11 [stack_trace]
- meta 1.7.0
- path 1.8.0
- plugin_platform_interface 2.0.2 [meta]
- sky_engine 0.0.99
- source_span 1.8.1 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.4.3 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0 [collection]
- vector_math 2.1.1
```

</details>

---

**Comment by hazzo** [comment]
Maybe this is related to this PR [7610](https://github.com/FirebaseExtended/flutterfire/pull/7610)

**Comment by darshankawar** [comment]
@hazzo 
The error log you shared also has this entry:

`8.9.0 - [Firebase/Core][I-COR000003] The default Firebase app has not yet been configured. Add `FirebaseApp.configure()` to your application initialization. This can be done in in the App Delegate's application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI). Read more: https://goo.gl/ctyzm8.`

Indicating  `firebase_core` not initialized. Can you confirm this by omitting step 3 and 6 and then have step 7 and run to see what you get ?

**Comment by hazzo** [comment]
@darshankawar Sorry I deleted the example repo and changed my project to Firebase Manual Initialization. 

From what I can remember the message of firebase not being initialized happened only when using fcm and setting iOS capabilities but the combination of this and auth made the app crash. 

I think you should review all the firebase plug-in use cases with Dart initialization because it's kind of buggy. As I already said in another issue Dart initialization should be stated in the documentation of flutter fire as a "experimental" way to initialize firebase. Is far from working okey.

**Comment by darshankawar** [comment]
@hazzo 
I used the steps to replicate using latest stable and ran the setup on iOS simulator which compiled and built the app properly:

```
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
Running Xcode build...
Xcode build done.                                           36.7s
Debug service listening on ws://127.0.0.1:52713/VoqxpZwyttw=/ws
Syncing files to device iPhone 12 Pro Max...
```

```
import 'firebase_options.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );
  runApp(const CupertinoExample());
}
```

  ```
firebase_core: ^1.10.6
  firebase_messaging: ^11.2.4
  firebase_auth: ^3.3.4
```

<img width="340" alt="Screenshot 2022-01-07 at 5 48 31 PM" src="https://user-images.githubusercontent.com/67046386/148543126-b84a959a-7379-4bbf-8287-d1faaa29917a.png">

Is there any specific background mode you selected ?

**Comment by hazzo** [comment]
I have to say that this only happens on real device, as this [previous issue](https://github.com/FirebaseExtended/flutterfire/issues/7659).

The background modes are the ones stated in flutter fire docs:

> Now ensure that both the "Background fetch" and the "Remote notifications" sub-modes are enabled:

**Comment by hazzo** [comment]
FCM would never work on an iOS simulator through APNs so I always test on a real device.

From the docs:
> IOS SIMULATORS

FCM via APNs does not work on iOS Simulators. To receive messages & notifications a real device is required.

**Comment by darshankawar** [comment]
Yes, that's correct. I tried to run the same scenario on physical device, but getting some another issue due to which I am unable to replicate this (error: unable to install app), so keeping it open and labeling for further input from the team.

/cc @Salakar

**Comment by bretmh** [comment]
I'm experiencing this once I've enabled capabilities **Push Notifications** and **Background Modes**. I'm running on a physical device of course to test push notifications on iOS.

Background Modes selected are **Background fetch** and **Remote notifications**. Both are specified in the documentation provided here https://firebase.flutter.dev/docs/messaging/apple-integration/#configuring-your-app.

<details>
<summary>notes</summary>

Disabling both of the capabilities the app builds and runs as expected. Additionally, when only adding the Push Notifications capability I get a very similar error.
```
2022-01-11 12:55:47.403262-0800 Runner[3868:1510125] Metal API Validation Enabled
2022-01-11 12:55:47.505159-0800 Runner[3868:1510125] Writing analzed variants.
2022-01-11 12:55:47.514670-0800 Runner[3868:1510368] 8.9.0 - [Firebase/Core][I-COR000012] Could not locate configuration file: 'GoogleService-Info.plist'.
2022-01-11 12:55:47.517533-0800 Runner[3868:1510125] FBSDKLog: FBSDKGraphRequestConnection cannot be started before Facebook SDK initialized.
2022-01-11 12:55:47.527883-0800 Runner[3868:1510125] FBSDKLog: Missing [FBSDKAppEvents appID] for [FBSDKAppEvents publishATE:]
2022-01-11 12:55:47.528576-0800 Runner[3868:1510125] <Warning>: Please set a value for FacebookAutoLogAppEventsEnabled. Set the flag to TRUE if you want to collect app install, app launch and in-app purchase events automatically. To request user consent before collecting data, set the flag value to FALSE, then change to TRUE once user consent is received. Learn more: https://developers.facebook.com/docs/app-events/getting-started-app-events-ios#disable-auto-events.
2022-01-11 12:55:47.528611-0800 Runner[3868:1510125] <Warning>: You haven't set a value for FacebookAdvertiserIDCollectionEnabled. Set the flag to TRUE if you want to collect Advertiser ID for better advertising and analytics results.
2022-01-11 12:55:47.529364-0800 Runner[3868:1510125] FBSDKLog: Starting with v13 of the SDK, a client token must be embedded in your client code before making Graph API calls. Visit https://developers.facebook.com/docs/ios/getting-started#step-3---configure-your-project to learn how to implement this change.
2022-01-11 12:55:47.739908-0800 Runner[3868:1510125] FBSDKLog: Missing [FBSDKAppEvents appID] for [FBSDKAppEvents publishInstall:]
2022-01-11 12:55:47.802058-0800 Runner[3868:1510125] FBSDKLog: Missing [FBSDKAppEvents appEventsState.appID] for [FBSDKAppEvents flushOnMainQueue:]
2022-01-11 12:55:47.802180-0800 Runner[3868:1510125] FBSDKLog: Missing [FBSDKAppEvents appEventsState.appID] for [FBSDKAppEvents flushOnMainQueue:]
2022-01-11 12:55:47.802246-0800 Runner[3868:1510125] FBSDKLog: Missing [FBSDKAppEvents appEventsState.appID] for [FBSDKAppEvents flushOnMainQueue:]
2022-01-11 12:55:47.822779-0800 Runner[3868:1510125] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The default FirebaseApp instance must be configured before the default Authinstance can be initialized. One way to ensure this is to call `FirebaseApp.configure()` in the App Delegate's `application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI).'
*** First throw call stack:
(0x181a5005c 0x199f6af54 0x181aa7190 0x1051618a0 0x1048bebec 0x10cdbe9a8 0x184dbf21c 0x105fa8a20 0x105faa700 0x105fbaa90 0x181a08ce4 0x1819c2ebc 0x1819d63c8 0x19d1e738c 0x18437c0bc 0x1840f9be8 0x1048a13cc 0x104be5a24)
libc++abi: terminating with uncaught exception of type NSException
dyld4 config: DYLD_LIBRARY_PATH=/usr/lib/system/introspection DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The default FirebaseApp instance must be configured before the default Authinstance can be initialized. One way to ensure this is to call `FirebaseApp.configure()` in the App Delegate's `application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI).'
terminating with uncaught exception of type NSException
(lldb) 
```
</details>


<details>
<summary>flutter dependencies</summary>

```
flutter pub deps -- --style=compact
Dart SDK 2.15.1
Flutter SDK 2.8.1
safe_entry 1.0.0+1

dependencies:
- authy 0.2.0 [flutter flutterfire_ui flutter_bloc firebase_core firebase_auth cloud_functions cloud_firestore equatable get_it esp_blufi mockito firebase_messaging]
- cloud_firestore 3.1.6 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- cloud_functions 3.2.3 [cloud_functions_platform_interface cloud_functions_web firebase_core firebase_core_platform_interface flutter]
- cupertino_icons 1.0.4
- esp_blufi 1.1.0 [equatable flutter_reactive_ble flutter]
- firebase_core 1.10.6 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_bloc 8.0.0 [flutter bloc provider]
- flutter_launcher_icons 0.9.2 [args image path yaml]
- flutter_reactive_ble 4.0.1 [collection flutter functional_data meta pedantic reactive_ble_mobile reactive_ble_platform_interface]
- permission_handler 8.3.0 [flutter meta permission_handler_platform_interface]
- syncfusion_flutter_sliders 19.4.38 [flutter intl syncfusion_flutter_core]

dev dependencies:
- flutter_lints 1.0.4 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- _fe_analyzer_shared 31.0.0 [meta]
- analyzer 2.8.0 [_fe_analyzer_shared cli_util collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- archive 3.1.6 [crypto path]
- args 2.3.0
- async 2.8.2 [collection meta]
- bloc 8.0.1 [meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- build 2.2.0 [analyzer async convert crypto glob logging meta path]
- built_collection 5.1.1
- built_value 8.1.3 [built_collection collection fixnum meta]
- characters 1.2.0
- charcode 1.3.1
- cli_util 0.3.5 [meta path]
- clock 1.1.0
- cloud_firestore_platform_interface 5.4.11 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 2.6.6 [cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- cloud_functions_platform_interface 5.0.18 [firebase_core flutter meta plugin_platform_interface]
- cloud_functions_web 4.2.4 [cloud_functions_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- code_builder 4.1.0 [built_collection built_value collection matcher meta]
- collection 1.15.0
- convert 3.0.1 [typed_data]
- crypto 3.0.1 [collection typed_data]
- dart_style 2.2.1 [analyzer args path pub_semver source_span]
- desktop_webview_auth 0.0.2 [crypto flutter http]
- equatable 2.0.3 [collection meta]
- fake_async 1.2.0 [clock collection]
- file 6.1.2 [meta path]
- firebase_auth 3.3.4 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_auth_platform_interface 6.1.9 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 3.3.5 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.2.3 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.5.3 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_database 9.0.5 [firebase_core firebase_core_platform_interface firebase_database_platform_interface firebase_database_web flutter]
- firebase_database_platform_interface 0.2.0+4 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_database_web 0.2.0+4 [firebase_core firebase_core_web firebase_database_platform_interface flutter flutter_web_plugins js]
- firebase_dynamic_links 4.0.4 [firebase_core firebase_core_platform_interface firebase_dynamic_links_platform_interface flutter meta plugin_platform_interface]
- firebase_dynamic_links_platform_interface 0.2.0+4 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging 11.2.4 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_messaging_platform_interface 3.1.4 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 2.2.5 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- fixnum 1.0.0
- flutter_facebook_auth 3.5.7 [flutter flutter_facebook_auth_platform_interface flutter_facebook_auth_web]
- flutter_facebook_auth_platform_interface 2.7.1 [flutter plugin_platform_interface]
- flutter_facebook_auth_web 2.6.0+2 [flutter flutter_web_plugins js flutter_facebook_auth_platform_interface]
- flutter_localizations 0.0.0 [flutter intl characters clock collection meta path typed_data vector_math]
- flutter_svg 1.0.0 [flutter meta path_drawing vector_math xml]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- flutterfire_ui 0.3.0 [cloud_firestore collection crypto desktop_webview_auth firebase_auth firebase_core firebase_database firebase_dynamic_links flutter flutter_facebook_auth flutter_localizations flutter_svg google_sign_in sign_in_with_apple twitter_login]
- functional_data 1.0.0 [meta collection]
- get_it 7.2.0 [async collection]
- glob 2.0.2 [async collection file path string_scanner]
- google_sign_in 5.2.2 [flutter google_sign_in_platform_interface google_sign_in_web]
- google_sign_in_platform_interface 2.1.1 [flutter quiver]
- google_sign_in_web 0.10.0+4 [flutter flutter_web_plugins google_sign_in_platform_interface js pedantic]
- http 0.13.4 [async http_parser meta path]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image 3.1.0 [archive meta xml]
- intl 0.17.0 [clock path]
- js 0.6.3
- lints 1.0.1
- logging 1.0.2
- matcher 0.12.11 [stack_trace]
- meta 1.7.0
- mockito 5.0.17 [analyzer build code_builder collection dart_style matcher meta path source_gen test_api]
- nested 1.0.0 [flutter]
- package_config 2.0.2 [path]
- path 1.8.0
- path_drawing 1.0.0 [vector_math meta path_parsing flutter]
- path_parsing 1.0.0 [vector_math meta]
- pedantic 1.11.1
- permission_handler_platform_interface 3.7.0 [flutter meta plugin_platform_interface]
- petitparser 4.4.0 [meta]
- plugin_platform_interface 2.0.2 [meta]
- protobuf 2.0.1 [fixnum collection]
- provider 6.0.1 [collection flutter nested]
- pub_semver 2.1.0 [collection meta]
- quiver 3.0.1+1 [matcher]
- reactive_ble_mobile 4.0.1 [flutter protobuf reactive_ble_platform_interface]
- reactive_ble_platform_interface 4.0.1 [collection flutter functional_data meta plugin_platform_interface]
- sign_in_with_apple 3.3.0 [flutter meta sign_in_with_apple_platform_interface sign_in_with_apple_web]
- sign_in_with_apple_platform_interface 1.0.0 [flutter plugin_platform_interface meta]
- sign_in_with_apple_web 1.0.1 [flutter flutter_web_plugins sign_in_with_apple_platform_interface js]
- sky_engine 0.0.99
- source_gen 1.2.1 [analyzer async build dart_style glob meta path source_span yaml]
- source_span 1.8.1 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- syncfusion_flutter_core 19.4.38 [vector_math flutter]
- term_glyph 1.2.0
- test_api 0.4.3 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- twitter_login 4.0.1 [flutter crypto http]
- typed_data 1.3.0 [collection]
- vector_math 2.1.1
- watcher 1.0.1 [async path]
- xml 5.3.1 [collection meta petitparser]
- yaml 3.1.0 [collection source_span string_scanner]
```
</details>

**Comment by rmkubala** [comment]
I managed to solve this issue by just adding GoogleService-Info.plist to the project (using XCode) along the dart initialization.

**Comment by hazzo** [comment]
The whole purpose of dart initialization is avoiding to do that.

**Comment by rmkubala** [comment]
I fully agree. I hope it's just a workaround and it will be fixed soon.

**Comment by bretmh** [comment]
> I managed to solve this issue by just adding GoogleService-Info.plist to the project (using XCode) along the dart initialization.

Is this confirmed? I will be testing it out and reporting my findings.

**Comment by anthonyhunter** [comment]
@darshankawar - the issue still exists for me after the fix.

Using a clean Flutter 2.8.1 project setup for firebase messaging, auth and firestore.
Run on a real device (iPhone 12 15.2.1)

With **Push Messaging** capability **enabled** I get the follow crash:

> *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The default FirebaseApp instance must be configured before the default Authinstance can be initialized. One way to ensure this is to call `FirebaseApp.configure()` in the App Delegate's `application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI).'

With **Push Messaging** capability **disabled** the app launches as expected.


<details><summary>Flutter doctor</summary>

```
[✓] Flutter (Channel stable, 2.8.1, on macOS 11.6.3 20G415 darwin-x64, locale en-AU)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.63.2)
[✓] Connected device (3 available)

• No issues found!
```
</details>

<details><summary>Flutter dependencies</summary>

```
Dart SDK 2.15.1
Flutter SDK 2.8.1
userapp 1.0.0+1

dependencies:
- cloud_firestore 3.1.7 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- firebase_app_check 0.0.6+5 [firebase_app_check_platform_interface firebase_app_check_web firebase_core firebase_core_platform_interface flutter]
- firebase_auth 3.3.6 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 1.12.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_messaging 11.2.6 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]

dev dependencies:
- flutter_lints 1.0.4 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- async 2.8.2 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- cloud_firestore_platform_interface 5.4.12 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 2.6.7 [cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.15.0
- fake_async 1.2.0 [clock collection]
- firebase_app_check_platform_interface 0.0.3+5 [firebase_core flutter meta plugin_platform_interface]
- firebase_app_check_web 0.0.5+5 [firebase_app_check_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_auth_platform_interface 6.1.11 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 3.3.7 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.2.4 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.5.4 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_messaging_platform_interface 3.1.6 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 2.2.7 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- intl 0.17.0 [clock path]
- js 0.6.3
- lints 1.0.1
- matcher 0.12.11 [stack_trace]
- meta 1.7.0
- path 1.8.0
- plugin_platform_interface 2.1.2 [meta]
- sky_engine 0.0.99
- source_span 1.8.1 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.4.3 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0 [collection]
- vector_math 2.1.1
```
</details>

**Comment by leidig54** [comment]
Same as above.

```
2022-01-30 14:42:48.182378+0000 Runner[842:41229] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The default FirebaseApp instance must be configured before the default Authinstance can be initialized. One way to ensure this is to call `FirebaseApp.configure()` in the App Delegate's `application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI).'
*** First throw call stack:
(0x18120d0fc 0x199a59d64 0x181264230 0x10560db84 0x1047dd840 0x1088de9a8 0x1845804e8 0x1063546d4 0x1063563b4 0x106366898 0x1811c5d84 0x18117ff5c 0x181193468 0x19cd3338c 0x183b365d0 0x1838b4f74 0x1045c5d24 0x104e05aa4)
libc++abi: terminating with uncaught exception of type NSException
dyld4 config: DYLD_LIBRARY_PATH=/usr/lib/system/introspection DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The default FirebaseApp instance must be configured before the default Authinstance can be initialized. One way to ensure this is to call `FirebaseApp.configure()` in the App Delegate's `application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI).'
terminating with uncaught exception of type NSException
(lldb) 
```



<details>
<summary>Flutter Doctor</summary>


```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-arm, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] IntelliJ IDEA Community Edition (version 2021.2.3)
[✓] VS Code (version 1.63.2)
[✓] Connected device (3 available)

• No issues found!
```
</details>

<details>
<summary>Dependencies</summary>


```environment:
  sdk: ">=2.15.0-268.18.beta <3.0.0"

dependencies:
  auto_size_text: ^3.0.0
  avatar_glow: ^2.0.2
  cloud_firestore: ^3.1.5
  convex_bottom_bar: ^3.0.0
  cupertino_icons: ^1.0.2
  dartx: ^1.0.0
  firebase_auth: ^3.3.4
  firebase_core: ^1.10.6
  firebase_dynamic_links: ^4.0.3
  firebase_messaging: ^11.2.6
  firebase_storage: ^10.2.6
  flutter:
    sdk: flutter
  flutter_keyboard_visibility: ^5.1.0
  flutter_launcher_icons: ^0.9.2
  flutter_slidable: ^1.2.0
  flutterfire_ui: ^0.3.0
  form_builder_validators: ^7.2.0
  image_picker: ^0.8.4+5
  material_floating_search_bar: ^0.3.6
  provider: ^6.0.1
  pull_to_refresh: ^2.0.0
  qr_code_scanner: ^0.6.1
  qr_flutter: ^4.0.0
  share_plus: ^3.0.4
  url_launcher: ^6.0.17

dev_dependencies:
  flutter_lints: ^1.0.0
  flutter_test:
    sdk: flutter
```
</details>

**Comment by anthonyhunter** [comment]
@darshankawar 
Would appear the fix for `firebase_messaging` #7610 should also be applied to `firebase_auth` as both packages require APNS token at launch.

**Comment by Shlaayl** [comment]
Will it be fixed soon?

**Comment by Shlaayl** [comment]
@anthonyhunter
Do I have to file a new report for this closed problem ?

**Comment by anthonyhunter** [comment]
> @anthonyhunter Do I have to file a new report for this closed problem ?

Started a new issue #7983

**Comment by walsha2** [comment]
> I managed to solve this issue by just adding GoogleService-Info.plist to the project (using XCode) along the dart initialization.

I can confirm that **this does indeed work** as a workaround (without needing to hand modify the package itself). 

I kept getting this error after adding `firebase_messaging` as a dependency. `firebase_auth` did not cause this issue, it was the addition of `firebase_messaging` when this all started. As soon as I added the `GoogleService-Info.plist`, with no other changes, the error was gone.

I obviously don't like using `GoogleService-Info.plist` as that is the purpose of the new Firebase Dart initialization 
as mentioned in another comment (https://github.com/FirebaseExtended/flutterfire/issues/7752#issuecomment-1014877987). Hopefully as soon as https://github.com/FirebaseExtended/flutterfire/issues/7983 is in and there is a new package release I can revert that GoogleService change on my end, but for now that is the workaround approach I am using as it is the least invasive.
